src/System.Security.SecureString/src/System/Security/SecureString.Windows.cs

```diff
@@ -0,0 +1,361 @@
+        }
+
+        private void EnsureCapacity(ref SafeBSTRHandle decryptedBuffer, int capacity)
+        {
+            if (capacity > MaxLength)
+            {
+                throw new ArgumentOutOfRangeException(nameof(capacity), SR.ArgumentOutOfRange_Capacity);
+            }
+
+            Debug.Assert(capacity > _decryptedLength, $"Expected to only be called when growing capacity = {capacity}, _decryptedLength = {_decryptedLength}");
+
+            SafeBSTRHandle newBuffer = SafeBSTRHandle.Allocate((uint)capacity);
+            SafeBSTRHandle.Copy(decryptedBuffer, newBuffer, (uint)_decryptedLength * sizeof(char));
+            decryptedBuffer.Dispose();
+            decryptedBuffer = newBuffer;
+        }
+
+        private void ProtectMemory(SafeBSTRHandle decryptedBuffer)
+        {
+            Debug.Assert(!decryptedBuffer.IsInvalid, "Invalid buffer!");
+
+            if (_decryptedLength == 0)
+            {
+                return;
+            }
+
+            try
+            {
+                SafeBSTRHandle newEncryptedBuffer;
+                if (!CryptProtectData(decryptedBuffer, out newEncryptedBuffer))
+                {
+                    throw new CryptographicException(Marshal.GetLastWin32Error());
+                }
+
+                _encryptedBuffer.Dispose();
+                _encryptedBuffer = newEncryptedBuffer;
+            }
+            finally
+            {
+                decryptedBuffer.ClearBuffer();
+                decryptedBuffer.Dispose();
+            }
+        }
+
+        private SafeBSTRHandle UnprotectMemory()
+        {
+            Debug.Assert(!_encryptedBuffer.IsInvalid, "Invalid buffer!");
+
+            if (_decryptedLength == 0)
+            {
+                return _encryptedBuffer;
+            }
+
+            SafeBSTRHandle decryptedBuffer;
+            if (!CryptUnprotectData(_encryptedBuffer, out decryptedBuffer))
+            {
+                throw new CryptographicException(Marshal.GetLastWin32Error());
+            }
+
+            return decryptedBuffer;
+        }
+
+        private const uint CRYPTPROTECTMEMORY_SAME_PROCESS = 0x00;
+
+        private static unsafe bool CryptProtectData(SafeBSTRHandle decryptedBuffer, out SafeBSTRHandle encryptedBuffer)
+        {
+            byte* uncryptedBufferPtr = null;
+            Interop.Crypt32.DATA_BLOB pDataOut = default(Interop.Crypt32.DATA_BLOB);
+            try
+            {
+                decryptedBuffer.AcquirePointer(ref uncryptedBufferPtr);
+
+                Interop.Crypt32.DATA_BLOB optionalEntropyBlob = default(Interop.Crypt32.DATA_BLOB);
+                Interop.Crypt32.DATA_BLOB pDataIn = new Interop.Crypt32.DATA_BLOB((IntPtr)uncryptedBufferPtr, decryptedBuffer.Length * 2);
+                if (Interop.Crypt32.CryptProtectData(ref pDataIn, String.Empty, ref optionalEntropyBlob, IntPtr.Zero, IntPtr.Zero, CRYPTPROTECTMEMORY_SAME_PROCESS, out pDataOut))
```

Comment:
Some streams have been crossed here.  Desktop calls advapi32!SystemFunction040 (RtlProtectMemory), whose analogue is CryptProtectMemory (which is where CRYPTPROTECTMEMORY_SAME_PROCESS is a documented value).  CPM is documented as not surviving a reboot.

CryptProtectData has different scopes.  The 0 we're passing here is just 0, it doesn't have a name.

CryptProtectMemory: Restricted to the current process, not supported in UWP.
CryptProtectData: Restricted to the current user, supported in UWP. (I'd guess UWP isolation makes "process, surviving reboot", but that's speculation).

If we want the same security level as desktop, we'll need to build OneCore and UWP differently.  Or, if we want to increase the read scope of OneCore we can...

Though, since it says "same user" for CPD and "same process" for CPM, we might want to try this with impersonation... if CPD doesn't work impersonated but CPM does, that would be a breaking change.